components/settings/actions/ActionDialog.js

```diff
@@ -31,7 +31,14 @@ export default class ActionDialog extends React.Component {
     performAction = () => {
         if (this.props.isFormValid === undefined || this.props.isFormValid()){
             GEPPETTO.trigger(GEPPETTO.Events.Show_spinner, this.props.message);
-            this.closeDialog();
+						this.closeDialog();
+						if (this.props.command == "netpyne_geppetto.deleteModel") {
```

Comment:
why this? what is the purpose of this bit of code?

Reply:
if you plot in the simulation tab, and then you  click **delete model** in the appbar, all the plots appear on top of the definition tab.  They don't get destroyed. I used to handle passing props to destroy the plots, but after the refactor, we are no longer sending props to the action, so I came up with this to overcome the issue

Reply:
I have moved these lines inside the 'then' clause along with the other similar actions. I am also deleting widget in a more elegant way now. https://github.com/MetaCell/geppetto-netpyne/pull/81/commits/18880bb404cec5d9393be5eafd4a89c141d55765